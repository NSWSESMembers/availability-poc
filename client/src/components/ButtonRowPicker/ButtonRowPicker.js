import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ModalSelector from 'react-native-modal-selector';


import { View } from 'react-native';
import { ButtonRow } from '../Button';

class ButtonRowPicker extends Component {
  onConfirm = (option) => {
    this.props.selected = option.label;
    this.props.onSelect(option.label);
  };

  render() {
    const data = [
      { key: 0, label: 'Available' },
      { key: 1, label: 'Unavailable' },
      { key: 2, label: 'Unavailable - unless urgent' },
    ];

    return (
      <View>
        <ModalSelector
          data={data}
          onChange={this.onConfirm}
        >
          <ButtonRow
            title={this.props.title}
            showIcon
            description={`For this time period I am '${this.props.selected}'`}
          />
        </ModalSelector>
      </View>
    );
  }
}

ButtonRowPicker.propTypes = {
  title: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
  selected: PropTypes.string.isRequired,
};

export default ButtonRowPicker;
