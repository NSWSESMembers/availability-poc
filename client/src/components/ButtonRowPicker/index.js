import ButtonRowPicker from './ButtonRowPicker';
import styles from './styles';

export { ButtonRowPicker };
