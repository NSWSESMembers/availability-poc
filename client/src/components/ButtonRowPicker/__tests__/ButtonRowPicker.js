import React from 'react';
import renderer from 'react-test-renderer';

import ButtonRowPicker from '../ButtonRowPicker';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ButtonRowPicker
      title="title"
      onSelect={noop}
      selected="selected"
      showIcon
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});

test('calls onConfirm', () => {
  let node;
  const component = renderer.create(
    <ButtonRowPicker
      title="title"
      onSelect={noop}
      selected="selected"
      showIcon
      ref={(ref) => { node = ref; }}
    />,
  );
  node.onConfirm({ label: 'label' });
  expect(component.toJSON()).toMatchSnapshot();
});
