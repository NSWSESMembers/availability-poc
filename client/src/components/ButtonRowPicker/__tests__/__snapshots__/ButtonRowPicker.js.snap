// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls onConfirm 1`] = `
<View>
  <View
    style={Object {}}
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "center",
              "padding": "5%",
            },
            Object {},
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "flexShrink": 1,
                "marginBottom": 8,
                "padding": 8,
              },
              Object {},
            ]
          }
        >
          <RCTScrollView
            keyboardShouldPersistTaps="always"
          >
            <View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Available
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Unavailable
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        Object {
                          "borderBottomWidth": 0,
                        },
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Unavailable - unless urgent
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "padding": 8,
                  },
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                cancel
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="none"
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "900",
                    "paddingTop": 5,
                  }
                }
              >
                TITLE
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingBottom": 5,
                  }
                }
              >
                For this time period I am 'selected'
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#EEE",
                    "fontSize": 40,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View>
  <View
    style={Object {}}
  >
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0,0,0,0.7)",
              "flex": 1,
              "justifyContent": "center",
              "padding": "5%",
            },
            Object {},
          ]
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255,255,255,0.8)",
                "borderRadius": 5,
                "flexShrink": 1,
                "marginBottom": 8,
                "padding": 8,
              },
              Object {},
            ]
          }
        >
          <RCTScrollView
            keyboardShouldPersistTaps="always"
          >
            <View>
              <View
                style={
                  Object {
                    "paddingHorizontal": 10,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Available
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Unavailable
                    </Text>
                  </View>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderBottomColor": "#ccc",
                          "borderBottomWidth": 1,
                          "padding": 8,
                        },
                        Object {},
                        Object {
                          "borderBottomWidth": 0,
                        },
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "rgba(0,118,255,0.9)",
                            "fontSize": 16,
                            "textAlign": "center",
                          },
                          Object {},
                        ]
                      }
                    >
                      Unavailable - unless urgent
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Array [
              Object {
                "alignSelf": "stretch",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "rgba(255,255,255,0.8)",
                    "borderRadius": 5,
                    "padding": 8,
                  },
                  Object {},
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#333",
                      "fontSize": 16,
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                cancel
              </Text>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        pointerEvents="none"
      >
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": 5,
                "paddingTop": 5,
              }
            }
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 11,
                    "fontWeight": "900",
                    "paddingTop": 5,
                  }
                }
              >
                TITLE
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingBottom": 5,
                  }
                }
              >
                For this time period I am 'selected'
              </Text>
            </View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#EEE",
                    "fontSize": 40,
                  },
                  undefined,
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
